etna/models/statsforecast.py

```diff
@@ -0,0 +1,795 @@
+
+    Method ``predict`` only works on in-sample data.
+    """
+
+    def __init__(
+        self,
+        order: Tuple[int, int, int] = (0, 0, 0),
+        season_length: int = 1,
+        seasonal_order: Tuple[int, int, int] = (0, 0, 0),
+        **kwargs,
+    ):
+        """Init model with given params.
+
+        Parameters
+        ----------
+        order:
+            A specification of the non-seasonal part of the ARIMA model: the three components (p, d, q) are the AR order, the degree of differencing, and the MA order.
+        season_length:
+            Number of observations per unit of time. Ex: 24 Hourly data.
+        seasonal_order:
+            A specification of the seasonal part of the ARIMA model.
+            (P, D, Q) for the  AR order, the degree of differencing, the MA order.
+        **kwargs:
+            Additional parameters for :py:class:`statsforecast.models.ARIMA`.
+        """
+        self.order = order
+        self.season_length = season_length
+        self.seasonal_order = seasonal_order
+        self.kwargs = kwargs
+        super().__init__(
+            base_model=_ARIMAAdapter(
+                order=self.order,
+                season_length=self.season_length,
+                seasonal_order=self.seasonal_order,
+                **self.kwargs,
+            ),
+        )
+
+    def params_to_tune(self) -> Dict[str, BaseDistribution]:
+        """Get default grid for tuning hyperparameters.
+
+        This grid tunes parameters: ``order.0``, ``order.1``, ``order.2``.
+        If ``self.season_length`` is greater than one, then it also tunes parameters:
+        ``seasonal_order.0``, ``seasonal_order.1``, ``seasonal_order.2``.
+        Other parameters are expected to be set by the user.
+
+        Returns
+        -------
+        :
+            Grid to tune.
+        """
+        grid: Dict[str, BaseDistribution] = {
+            "order.0": IntDistribution(low=1, high=6),
+            "order.1": IntDistribution(low=1, high=2),
+            "order.2": IntDistribution(low=1, high=6),
+        }
+
+        num_periods = self.season_length
+        if num_periods > 1:
+            grid.update(
+                {
+                    "seasonal_order.0": IntDistribution(low=0, high=2),
+                    "seasonal_order.1": IntDistribution(low=0, high=1),
+                    "seasonal_order.2": IntDistribution(low=0, high=1),
+                }
+            )
+
+        return grid
+
+
+class StatsForecastAutoThetaModel(
+    PerSegmentModelMixin, PredictionIntervalContextIgnorantModelMixin, PredictionIntervalContextIgnorantAbstractModel
+):
+    """
+    Class for holding :py:class:`statsforecast.models.AutoTheta`.
+
+    Method ``forecast`` only works on ouf-of-sample data that goes right after training data.
+
+    Method ``predict`` only works on in-sample data.
+    """
+
+    def __init__(
+        self,
+        season_length: int = 1,
+        decomposition_type: str = "multiplicative",
```

Comment:
We can use Lateral type here

Reply:
Here I used the type from `statsforecast` library. For this model the possible values are: `[None, "STM", "OTM", "DSTM", "DOTM"]` according to the [code](https://github.com/Nixtla/statsforecast/blob/main/statsforecast/theta.py#L535).

I'm not sure if we should fix that in our code.

Reply:
There multiple ways to make it more obvious